### ISSUE TITLE
players may register with trailing underscore... then can't log in.

### ISSUE THREAD

**Issue by ExeterDad** [ShadowNinja/name_restrictions#4]
Twice now a player has created a alt that is the exact name as their regular name, but has a trailing underscore.

Example
----------------------
Regular:  Player
Alt:   Player_

Users are saying that they are unable to log in with alt name AND original name after registration.

Any thoughts?

**Comment by ExeterDad** [comment]
Correction.
Player joins with regular name --> Player
Alt with trailing underscore  --> Player_
Can no longer log in with --> Player

Message is: "Your name is too similar to another player's name."

**Comment by ShadowNinja** [comment]
I've fixed this two ways.
First, I fixed the root cause by stripping `-` and `_` from the name before the match: 1e6d9e01b78980f7cb372038b6e328b3ca9f17ad

Then, I prevented players from getting locked out by exempting players that already exist: 87ed4ade88c63fa64ead939d8051484544a57ac3  In addition to the malicious attack that you suggested, this also prevents name policy changes from affecting existing players.

Note that this could never disable the server admin's account -- they're specifically exempted from all name checks (it could however disable the account of a mod or other trusted user).
